vulcan/lib/client/jsx/selectors/workflow_selectors.ts

```diff
@@ -294,14 +293,23 @@ export const sourceNameOfReference = (ref: [string, string]) => {
 };
 
 export const sourceNamesOfStep = (step: WorkflowStep) => {
-  return step.out.map((name) => sourceNameOfReference([step.name, name]));
-};
+  if (stepCollapsesOutputs(step))
+    // Assign one output here so that any downloaded input data is grabbed.
+    return [sourceNameOfReference([step.name, step.out[0]])];
```

Comment:
Here I "hide" any multiple outputs for a single CWL step. If you don't "hide" the extra outputs, then you get duplicate UI components rendered on the screen. For example, when I added two outputs to the `transform_data` step, I got two "DataTransformationInput" components, each of which had a button and a dialog. And unfortunately one dialog would be tied to `data`, and the other tied to `source_data`, which isn't the desired behavior...